xwiki-commons-core/xwiki-commons-velocity/src/main/java/org/xwiki/velocity/tools/URLTool.java

```diff
@@ -58,4 +60,19 @@
         }
         return queryParams;
     }
+
+    /**
+     * Convert a string to an URL object, if it's possible.
+     *
+     * @param urlString string to be convert
+     * @return an URL object from a given string
+     */
+    public URL getURL(String urlString)
```

Comment:
I feel like the method should be called `parseURL` since that's what it's really about.